### PR TITLE
Bugfix/set isrelease for installer releases

### PR DESCRIPTION
**Pull request by eventuri**
This merge request has code that sets `-PisRelease`, along with `-Pfinal` or `-Prc`, for the `.base-release-installer` Gitlab job. 

By setting -PisRelease, build.gradle files can leverage the BSP's `isRelease` variable to conditionally publish to specific `custom` snapshot or `release` repos. For example the below would be in a build.gradle file. 

```
buildSupport {
    repoManagement {
        custom name: 'example-Release', url: 'https://example.com:443/artifactory/example-releases', credentialName: 'EXAMPLE', publish: isRelease
        custom name: 'example-Snapshot', url: 'https://example.com:443/artifactory/example-snapshots', credentialName: 'EXAMPLE', publish: !isRelease
   }
}
```

Note, this merge request heavily borrows from the merge request at https://github.com/chesapeaketechnology/gitlab-templates/pull/74, thanks @AndrewTrow! I went ahead and closed https://github.com/chesapeaketechnology/gitlab-templates/pull/74 since it has the same purpose as this merge request.

### CODE REVIEW

File: lib/gitlab/ci/templates/pipeline/GradleInstall4JPipeline.yml [hunk 1]
```diff
@@ -59,6 +56,19 @@ snapshot-installer:
 
 .base-release-installer:
   extends: .base-installer
+  before_script:
+    - !reference [ .base-installer, before_script ]
+    # Determine appropriate gradle flags for release or release candidate
+    - >
+      if [[ $(echo $RELEASE | tr '[:upper:]' '[:lower:]') == "final" ]] ; then
+      echo "FINAL build";
```
Comment (lstreepy) [inline lib/gitlab/ci/templates/pipeline/GradleInstall4JPipeline.yml]: nit: please add indentation to make this easier to read

**Comment by lstreepy** [comment]
What project needed the inclusion of `-Pfinal` or `-Prc` and how are they used?  The ISP doesn't expect either `-Prc` or `-Pfinal`, it looks for `-Psnapshot`.  This is a major breaking change to the pipeline (since it affects what the user has to enter in web UI when running the pipeline).  Or, maybe it isn't a problem if we end up with `-Prc=true`??  At a minimum, I would ask that this NOT be merged into 1.3, and create 1.4 instead.

**Comment by eventuri** [comment]
> 



> 
@lstreepy The  `-Pfinal` and `-Prc` is for @JRhoades22 project. 

The thought behind this merge request was that by adding ` -Pfinal` and `Prc` to the `.base-release-installer:` Gitlab job, that @JRhoades22 team could choose to release a final installer or release candidate installer via the Gitlab web UI. 
However thinking about it more, I could just trash this merge request and instead have @JRhoades22 team set their release candidate installer version via their installer's projectVersion.txt. Let me talk to @JRhoades22 and see what works best for them.

**Comment by lstreepy** [comment]
That is the intended design.  If you specify `-Prelease=<anything>`, then the installer is built using the version in the `projectVersion.txt` file.  The intent being that you modify that file according to the type of release you want (RC or FINAL).  If you look at the TAKX CI build, when we do a release build, it updates the `raptorVersion.txt` file in `takx-installer` accordingly, and kicks off a release build.

**Comment by eventuri** [comment]
> Makes sense, I decided to not modify the installer pipeline Gitlab template and go another route.

**Pull request closed without merging**